mobile/test/common/integration/engine_with_test_server.cc

```diff
@@ -0,0 +1,25 @@
+#include "test/common/integration/engine_with_test_server.h"
+
+namespace Envoy {
+
+EngineWithTestServer::EngineWithTestServer(Platform::EngineBuilder& engine_builder,
+                                           TestServerType type) {
+  test_server_.startTestServer(type);
+  engine_ = engine_builder.build();
+}
+
+EngineWithTestServer::~EngineWithTestServer() {
+  engine_->terminate();
+  // It is important that the we shutdown the TestServer first before terminating the Engine. This
```

Comment:
The comment says we should shutdown the test server first, then terminate the engine, but the destructor is doing the opposite

Reply:
I'm an idiot :( Nice catch.